src/Parser/Hunter/BeastMastery/Modules/Traits/TitansThunder.js

```diff
@@ -0,0 +1,132 @@
+  };
+
+  goodTTCasts = 0;
+  badTTCasts = 0;
+  _currentStacks = 0;
+  totalTTCasts = 0;
+  stacksOnTTCast = 0;
+  shouldHaveSavedTT = 0;
+
+  on_toPlayer_applybuff(event) {
+    const buffId = event.ability.guid;
+    if (buffId !== SPELLS.DIRE_BEAST_BUFF.id) {
+      return;
+    }
+    this._currentStacks += 1;
+  }
+  on_toPlayer_removebuff(event) {
+    const buffId = event.ability.guid;
+    if (buffId !== SPELLS.DIRE_BEAST_BUFF.id) {
+      return;
+    }
+    this._currentStacks -= 1;
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.TITANS_THUNDER.id) {
+      return;
+    }
+    debug && console.log(`stacks:`, this._currentStacks);
+    this.totalTTCasts += 1;
+    if (this.spellUsable.cooldownRemaining(SPELLS.BESTIAL_WRATH.id) < 30) {
+      this.shouldHaveSavedTT += 1;
+      this.badTTCasts += 1;
+      return;
+    }
+    if (!this.combatants.selected.hasTalent(SPELLS.DIRE_FRENZY_TALENT.id)) {
+      if (this.combatants.selected.hasBuff(SPELLS.DIRE_BEAST_BUFF.id)) {
+        this.goodTTCasts += 1;
+        this.stacksOnTTCast += this._currentStacks;
+      } else {
+        this.badTTCasts += 1;
+      }
+    } else {
+      if (this.combatants.selected.hasBuff(SPELLS.BESTIAL_WRATH.id)) {
+        debug && console.log(`in good tt casts`);
+        this.goodTTCasts += 1;
+      } else {
+        this.badTTCasts += 1;
+      }
+    }
+  }
+  statistic() {
+    const averageDBPrTT = this.stacksOnTTCast / this.totalTTCasts;
+
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.TITANS_THUNDER.id} />}
+        value={(
+          <span>
+            {averageDBPrTT.toFixed(2)}{'  '}
+            <SpellIcon
+              id={SPELLS.DIRE_BEAST_BUFF.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+              }}
+            />
+            <br />
+            {this.goodTTCasts}{'  '}
+            <SpellIcon
+              id={SPELLS.TITANS_THUNDER.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+              }}
+            />
+            {'  '}
+            {this.badTTCasts}{'  '}
+            <SpellIcon
+              id={SPELLS.TITANS_THUNDER.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+                filter: 'grayscale(100%)',
+              }}
+            />
+          </span>
+
+        )}
+        label={`Titan's Thunder information`}
+        tooltip={`You cast Titan's Thunder ${this.totalTTCasts} times, of which ${this.badTTCasts} were bad casts. <br/> Bad casts indicate that they were used without Dire Beasts up, or if using Dire Frenzy it was used without Bestial Wrath up. <br/> You cast Titan's Thunder ${this.shouldHaveSavedTT} times where you should have delayed casting it, this occurs when you cast Titan's Thunder when there is less than 30 seconds remaning on Bestial Wrath cooldown.`}
```

Comment:
Check your wording: ` Bad casts indicate that they were used without Dire Beasts up, or if using Dire Frenzy it was used without Bestial Wrath up.`